ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -1002,36 +1002,46 @@ let assert_can_migrate  ~__context ~vm ~dest ~live ~vdi_map ~vif_map ~options =
       `cross_pool
   in
 
-  (* Check VDIs are not migrating to or from an SR which doesn't have required_sr_operations *)
+  (* operations required for migration *)
   let required_sr_operations = [Smint.Vdi_mirror; Smint.Vdi_snapshot] in
-  assert_sr_support_operations ~__context ~vdi_map ~remote ~ops:required_sr_operations;
```

Comment:
Why have you replaced this single call to `assert_sr_support_operations` with two identical calls, one for each `migration_type`? It looks deliberate but I don't see why you would do it. (Perhaps to rearrange the order of the assertions/tests?)